aten/src/ATen/templates/TypeDefault.h

```diff
@@ -12,9 +12,15 @@ struct AT_API TypeDefault : public Type {
 
   // Make sure overload resolution considers the nullary virtual method.
   // (A single argument overload is generated in the list.)
-  bool is_cuda() const override = 0;
-  bool is_sparse() const override = 0;
-  bool is_distributed() const override = 0;
+  bool is_cuda() const override {
+    return backend() == Backend::CUDA || backend() == Backend::SparseCUDA;
+  }
+  bool is_sparse() const override {
+    return backend() == Backend::SparseCPU || backend() == Backend::SparseCUDA;
+  }
+  bool is_distributed() const override {
```

Comment:
maybe (eventually) add `is_complex` here too?

Reply:
Perhaps. Actually, since `Type` isn't really supposed to be a public API anymore, I might advocate more for adding this directly to `Tensor` itself.

Reply:
Ah okay.